lib/dogma/rule/taken_name.ex

```diff
@@ -21,59 +21,102 @@ defrule Dogma.Rule.TakenName do
     end
   """
 
-  reserved_words = ~w(
+  other_reserved_words = ~w(
     __CALLER__ __DIR__ __ENV__ __MODULE__ abs alias apply bc binary_part
     binding bit_size byte_size case cond def defdelegate defexception defimpl
     defmacro defmacrop defmodule defoverridable defp defprotocol defstruct
     destructure div elem exit for function get_and_update_in get_in hd if
-    import inbits inli inlist inspect is_atom is_binary is_bitstring is_boolean
+    import inbits inli inlist is_atom is_binary is_bitstring is_boolean
     is_exception is_float is_function is_integer is_list is_map is_nil
     is_number is_pid is_port is_record is_reference is_tuple lc length make_ref
     map_size max min ast put_elem put_in quote raise receive rem require
     reraise respawn resque round self send sigil_C sigil_R sigil_S sigil_W
     sigil_c sigil_r sigil_s sigil_w spawn spawn_link spawn_monitor struct sum
-    super throw tl to_char_list to_string trunc try tuple_size unless unquote
+    super throw tl to_char_list trunc try tuple_size unless unquote
     unquote_splicing update_in use
   )a
 
-  @reserved_words Enum.into(reserved_words, MapSet.new)
+  @other_reserved_words Enum.into(other_reserved_words, MapSet.new)
+
+  @protocol_with_reserved_words %{
+    [:Inspect] => :inspect,
+    [:List, :Chars] => :to_charlist,
+    [:String, :Chars] => :to_string
+  }
 
   @spec reserved_words :: MapSet.t
   def reserved_words do
-    @reserved_words
+    @protocol_with_reserved_words
+    |> Map.values()
+    |> Enum.into(@other_reserved_words)
+  end
```

Comment:
How about we turn this into a module attribute? Saves us computing this value every time it is used.